rust/src/api/get_node_by_path.rs

```diff
@@ -65,19 +177,42 @@ mod test {
             ),
         ];
 
-        for test_case in tests {
-            let path = test_case.0;
-            let expected = test_case.1;
-            let actual = get_node_by_path(path);
-            assert_eq!(actual, expected);
+        for tc in tests {
+            let path = tc.0;
+            let expected = tc.1;
+            #[cfg(feature = "const")]
+            {
+                let actual = get_node_by_path(path);
+                assert_eq!(actual, expected);
+            }
+            #[cfg(feature = "storage")]
+            {
+                let actual = get_node_by_path("../_data", path).ok();
+                assert_eq!(actual, expected);
+            }
             #[cfg(feature = "serde_derive")]
             {
-                let expected_stringified = test_case.2;
+                let expected_stringified = tc.2;
+                let actual = get_node_by_path("../_data", path).ok();
```

Comment:
we can remove the inner scopes here, and use the `actual` from the same scope, conditionally declared for each feature.